File: src/main/java/pl/mroz/buddiesapi/domain/account/AccountRepository.java
package pl.mroz.buddiesapi.domain.account;

import pl.mroz.buddiesapi.infrastructure.database.location.LocationEntity;

import java.util.List;
import java.util.UUID;

public interface AccountRepository {

    List<Account> findAll();

    Account save(Account account);

    void delete(Account account);

    Account getById(UUID accountUUID);

    interface IAccountEntity {
        UUID getAccountId();

        String getEmail();

        String getHashedPassword();

        String getName();

        String getLastName();

        LocationEntity getLocationEntity();

    }
}

File: src/main/kotlin/buddiesapi/domain/account/AccountRepository.kt
package buddiesapi.domain.account

import buddiesapi.infrastructure.database.location.LocationEntity
import java.util.UUID

interface AccountRepository {
    fun findAll(): List<Account>

    fun save(account: Account): Account

    fun delete(account: Account)

    fun getById(accountUUID: UUID): Account?

    interface IAccountEntity {
        val accountId: UUID?

        val email: String

        val hashedPassword: String

        val name: String

        val lastName: String

        val locationEntity: LocationEntity
    }
}
